### PR TITLE
Cirrus CI: set working dir to be Go-compatible

### PR DESCRIPTION
**Pull request by orangecms**
Let's see if this works. :)

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -1,9 +1,27 @@
 freebsd_instance:
   image_family: freebsd-12-0
 test_task:
+  # see https://cirrus-ci.org/guide/tips-and-tricks/#custom-clone-command
+  # for Golang, we need to have a src/owner/project structure
+  clone_script: |
+    export CIRRUS_WORKING_DIR=$CIRRUS_WORKING_DIR/src/${CIRRUS_REPO_FULL_NAME}
```
Comment (rminnich) [inline .cirrus.yml]: for a lot of this, are you sure you can't just use go get?

Comment (orangecms) [reply .cirrus.yml]: I have no clue - I'll gladly try it out. :)

**Comment by orangecms** [comment]
Alright, this worked thus far: https://cirrus-ci.com/task/4880209009967104

And I took this part
```
      - run:
          name: Check vendored dependencies
          command: |
            ./dep version
            ./dep status
            ./dep ensure -vendor-only
            git status
            if [[ -n "$(git status --porcelain vendor)" ]]; then
              echo 'vendor/ is out-of-date: run `dep ensure -vendor-only` and then check in the changes'
              echo 'If `dep ensure` results in no changes, make sure you are using the latest relase of dep'
              git status --porcelain vendor
              exit 1
            fi
```
from `.circleci/config.yml`, which yields out of sync and missing packages here: https://cirrus-ci.com/task/5373660889612288

> Gopkg.lock is out of sync with imports and/or Gopkg.toml. Run `dep check` for details.

Now I've tried to simply run `go get ./...` as per https://cirrus-ci.org/examples/#go to see if that simply replaces the extensive mkdir/clone/GOPATH part. :crossed_fingers:

**Comment by orangecms** [comment]
How very unfortunate. https://cirrus-ci.com/task/5149083089502208

So I'll stick with the mkdir workaround solution. :woman_shrugging: 

I'll rebase my FreeBSD port fix attempt and see if it happily builds in the :cloud:. :)

**Pull request closed without merging**